### ISSUE TITLE
Some problems in using int8 calibration

### ISSUE THREAD

**Issue by 992355092** [NVIDIA/TensorRT#2036]
## Description

 I  generate my own cache file to TensorRT's use for calibration. and run model on two devices(2080ti and Xavier) ,They have the same tensorrt's  version(8.0.2),  but the results of network output are very different, 2080ti's result is the same as fp32 result, but Xavier's  accuracy is poor, I try to set the layer dynamic range to be consistent by Api tensor->setDynamicRange(min, max ), but output of 2080ti and Xavier results are still different，How can I make their outputs the same? 
## Environment

**TensorRT Version**: 8.0.2
**NVIDIA GPU**: 2080ti and Xavier
**NVIDIA Driver Version**: 
**CUDA Version**: 10.2
**CUDNN Version**: 8.0
**Operating System**: ubuntu 
**Python Version (if applicable)**: 
**Tensorflow Version (if applicable)**: 
**PyTorch Version (if applicable)**: 
**Baremetal or Container (if so, version)**:

**Comment by 992355092** [comment]
@zerollzeng this is my cache, I used it on 2080ti, and it's work well, but the same cache in Xavier, the network output is different from 2080ti
TRT-8203-EntropyCalibration2
167: 3c009976
168: 3e5d05cc
169: 3c00f9f4
170: 3d925bb9
171: 3eb0e47d
172: 3c00f9f4
173: 3e297d3a
174: 3e05400d
175: 3c00f9f4
176: 3d43b9a8
177: 3e0bf43e
178: 3c00f9f4
179: 3d6b8351
180: 3def4d0e
181: 3c00f9f4
182: 3d953b38
183: 3de56633
184: 3e15965b
185: 3c00f9f4
186: 3d909b75
187: 3dc0f5b7
188: 3de1551a
189: 3c00f9f4
190: 3d46aceb
191: 3d87bf3f
192: 3c00f9d2
193: 3ce6c4a3
194: 3d59de6b
195: 3bf2a0b9
196: 3c72776e
197: 3d44a1f3
198: 3c0068a8
199: 3d257a70
200: 3d2e18eb
201: 3c007abc
202: 3c48f4d7
203: 3cdacf6f
204: 3d536c45
205: 3c00e943
206: 3c8ff54c
207: 3d33282b
208: 3c00e4f9
209: 3ca57d89
210: 3d14f968
211: 3d3f77b9
212: 3c00a695
213: 3c52582f
214: 3d698b3c
215: 3c00f9de
216: 3d11bfe7
217: 3d70cc32
218: 3d838da4
219: 3c00e8b0
220: 3d1134a2
221: 3d42e8bc
222: 3d43e3d3
223: 3c00992b
224: 3c905118
225: 3d54c437
226: 3c00e9c4
227: 3cb468b3
228: 3d116fa8
229: 3befca73
230: 3be183c7
231: 3d83976c
232: 3c00f9f3
233: 3cd4088e
234: 3d2f55ab
235: 3bfeb39c
236: 3c31a813
237: 3cdb252f
238: 3d70d3e5
239: 3c00f9f1
240: 3c92a1ab
241: 3d608911
242: 3c00f951
243: 3ca606fd
244: 3d14d759
245: 3d7df03e
246: 3c00f9ed
247: 3c8826fa
248: 3d89b563
249: 3c00f9f4
250: 3d01909c
251: 3d6f63b5
252: 3d886261
253: 3c00f9e9
254: 3d123a18
255: 3d293721
256: 3d8a08c2
257: 3c00e9d2
258: 3c9221f6
259: 3d62519e
260: 3c00f932
261: 3ce90d20
262: 3d4a1d31
263: 3c00f97b
264: 3d0215e0
265: 3d84da2f
266: 3d8641f6
267: 3d865873
268: 3d6a1562
269: 3d5a9296
270: 3c00e344
271: 3cc00883
272: 3d450bb3
273: 3c00e7f9
274: 3cd3e818
275: 3d48418c
276: 3c00e6ac
277: 3cd29837
278: 3d4e949d
279: 3c00f988
280: 3cd301ba
281: 3d5f5f6a
282: 3c00e55e
283: 3cd29138
284: 3cd301ba
285: 3d5b840f
286: 3c00e72d
287: 3cd2ffd1
288: 3d544d6b
289: 3c00f917
290: 3cd38c8f
295: 3cd38c8f
296: 3cd45c3a
297: 3d5a1cdf
298: 3c00f948
299: 3ce71e49
300: 3d678a6b
301: 3c00f99f
302: 3d0122c9
303: 3d734fce
304: 3c00f9f4
305: 3d127ddc
306: 3d4d66ec
307: 3c00f98a
308: 3ce83eec
309: 3d026854
310: 3d73f694
311: 3c00f96a
312: 3d008834
313: 3d981997
314: 3c00f9f1
315: 3d005702
320: 3d005702
321: 3cd239e5
322: 3d7b3293
323: 3c00f984
324: 3d1068e3
325: 3d986aaf
326: 3c00f848
327: 3cd27234
328: 3d8d5869
329: 3c00f9f2
330: 3d018695
331: 3da8b3ba
332: 3c00f9ec
333: 3d254097
334: 3d118d4f
335: 3d9a77b6
336: 3c00f9f4
337: 3d250852
338: 3d8b69ea
339: 3c00f9dc
340: 3d110863
341: 3d005702
342: 3d7f087d
343: 3c00f9d8
344: 3ce9f822
345: 3d8591a3
346: 3c00f9e1
347: 3ce79796
348: 3d9fef9b
349: 3c00f9f4
350: 3d113789
351: 3d99c174
352: 3c00f9ee
353: 3d273563
354: 3d113789
355: 3d8d4960
356: 3c00f9f3
357: 3ce6cd00
358: 3d832e43
359: 3c00f9f3
360: 3d133407
361: 3d133407
362: 3d936284
363: 3c00f9ee
364: 3d01b174
365: 3d8bea7f
366: 3c00f9f1
367: 3d00a3fb
368: 3d90f808
369: 3c00f9f3
370: 3d12b2a7
371: 3d7cc88b
372: 3c00f909
373: 3ce8befd
374: 3d12b2a7
375: 3d82a405
376: 3c00f9e3
377: 3d018b35
378: 3e572a5e
390: 3e572a5e
391: 3e572a5e
392: 3bfaf6ed
434: 3e22f5ed
446: 3e22f5ed
447: 3e22f5ed
448: 3c00f081
490: 3e09ced2
502: 3e09ced2
503: 3e09ced2
504: 3c00efde

**Comment by nvpohanh** [comment]
Sometimes it is also related to which tactics and precisions are chosen on each GPU. Could you share the verbose logs when building the engines?

Also, I would suggest trying TRT 8.2 and/or 8.4EA if that is possible.

**Comment by zerollzeng** [comment]
Yes, I think it might be caused by some bad tactics in Xavier.

@992355092 can you try JetPack 5.0(TRT 8.4 EA) on Xavier first? if it still have problem, you can share your model(prefer onnx) to us.

**Comment by 992355092** [comment]
@nvpohanh  Hi, I used TRT 8.4 EA on orin, and build engine by the same calibration, it still have problem, 2080ti's precision is batter than orin,  there are verbose logs when building the engines
[x86_int8.log](https://github.com/NVIDIA/TensorRT/files/8946544/x86_int8.log)
[orin_int8.log](https://github.com/NVIDIA/TensorRT/files/8946545/orin_int8.log)

**Comment by nvpohanh** [comment]
@992355092 Could you re-generate the calibration cache using TRT 8.4? I see that the two logs have different fusion patterns, probably because they came from different TRT versions.

**Comment by nvpohanh** [comment]
also going back to the original question, for this specific model, the calibration cache cannot be shared between Xavier and 2080Ti because the fusion pattern will be different, even if the same TRT is used.

**Comment by nvpohanh** [comment]
Specifically, it is about whether Conv+Swish is fused. On Xavier, it is not fused, but it is on 2080Ti

**Comment by ttyio** [comment]
Closing since no response for more than 3 weeks, please reopen if you still have question, thanks!
